### DISCUSSION TITLE
General questions bout the compability and updates

### DISCUSSION THREAD

**Discussion by life8853** [Joshua-Riek/ubuntu-rockchip#158] [Q&A] [answered]
1. Will my Orange Pi 5 support everything that linux normally supports? (if i have dongle that works under my linux machine, will it work with this image too)
2. Can i upgrade all the stuff from this repo (firmware, patches etc.) just by apt or will i have to reflash the card someday? Cause backing everything up and redeploying it seems like a hassle.
3. Does Hardware Aceleration work in the Server image too, if I were to use some media streaming service like plex or jellyfin?
4. Also i am wondering between Opi 5 and Opi5b, im gonna use it with Home Assistant so the BT may come in handy someday, but i could also use a dongle ( i don't need BT right now). Any suggestions?

Big thanks!

**Comment by Joshua-Riek** [comment, marked as answer]
1. Sadly, not yet, we need to wait for the mainline Linux kernel to be compatible with the RK3588 SOC, I would expect this to be available in a stable capacity in about two years.
2. You can upgrade the kernel, firmware, and bootloader easily with apt as they are the core system components. However, other changes to the system will not be included in updates through apt, it would be extremely hard for me to implement as they generally relate to misc system services or pre-installed application improvements. 
3. Hardware acceleration is not included by default in the server image as it's minimal. However, the hardware acceleration can be installed manually with apt. Now I think about it, I should include the apt repos for hardware acceleration by default in the server image. As for plex and jellyfin I have no clue.
4. The RTL8821CU works great for WiFi and Bluetooth. However, I broke WiFi for the dongle in the last kernel update, I already have a fix, but it's not released yet.

**Reply by life8853** [reply]
So do i have a way of checking, if the usb device im willing to buy will work beforehand?

**Reply by Joshua-Riek** [reply]
Right now, no. The only available kernel that works is not standard, and specific USB WiFi drivers need to be hacked in by a developer. I'm working on adding more support, but it takes time. 

As mentioned above, I broke some WiFi drivers in the latest v1.15 release. I should push a fix out soon so devices like [this]( https://www.amazon.com/Bluetooth-Adapter-Wireless-External-Receiver/dp/B07YDFZWT8/ref=sr_1_1_sspa?adgrpid=1340305251782665&hvadid=83769292333310&hvbmt=be&hvdev=c&hvlocphy=72387&hvnetw=o&hvqmt=e&hvtargid=kwd-83769460321240%3Aloc-190&hydadcr=7283_13183180&keywords=rtl8821cu&qid=1684770998&sr=8-1-spons&psc=1&spLa=ZW5jcnlwdGVkUXVhbGlmaWVyPUExQTlOR1E1RzlVUVoyJmVuY3J5cHRlZElkPUEwMzU2NjIzM0ZWSkdQSjM3RFdZNCZlbmNyeXB0ZWRBZElkPUEwMTgyMDQxMzlFWklBQzdDNFdDWCZ3aWRnZXROYW1lPXNwX2F0ZiZhY3Rpb249Y2xpY2tSZWRpcmVjdCZkb05vdExvZ0NsaWNrPXRydWU=) will work again.

**Reply by life8853** [reply]
Okay, thanks @Joshua-Riek for all your great work for the community, i think ill go with orange pi anyway, since i won't need wifi or bt for a long time (I think)

**Comment by nobara256** [comment]
If you won't need to plug an M.2 device then 5B is a better choice as it comes with better WiFi & BT. But they are relatively expensive.
